### ISSUE TITLE
Layer (new component)

### ISSUE THREAD

**Issue by theetrain** [carbon-design-system/carbon-components-svelte#1892]
**Spun off from #1629**

- [ ] Add new Layer component. Reference: https://react.carbondesignsystem.com/?path=/docs/components-layer--overview
- [ ] Deprecate the `light` prop from `TextInput`. Recommend using `Layer` component instead. To be removed in `carbon-components-svelte@2.0.0`.
- [ ] Adjust related components to ensure they work well with Layer

**Comment by SimpleProgrammingAU** [comment]
I don't think the `light` prop should be removed in v11. The React library has it marked for removal in v12, doesn't it?

**Comment by theetrain** [comment]
You're right that the React component has it deprecated. We should follow a similar backwards-compatibility cadence:

https://github.com/carbon-design-system/carbon/blob/be888a850730d9658decaf1336f4b3175eb4d817/packages/react/src/components/TextInput/TextInput.tsx#L84-L90

I updated the criteria above.

**Comment by SimpleProgrammingAU** [comment]
What are the outstanding tasks following the merge of PR #1932?
